core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/CompactChangeEvent.java

```diff
@@ -0,0 +1,204 @@
+    asDoc.put(Fields.OPERATION_TYPE_FIELD, new BsonString(getOperationType().toRemote()));
+    asDoc.put(Fields.DOCUMENT_KEY_FIELD, getDocumentKey());
+
+    if (getFullDocument() != null && (getFullDocument() instanceof BsonValue)
+        && ((BsonValue) getFullDocument()).isDocument()) {
+      asDoc.put(Fields.FULL_DOCUMENT_FIELD, (BsonValue) getFullDocument());
+    }
+
+    if (getUpdateDescription() != null) {
+      asDoc.put(Fields.UPDATE_DESCRIPTION_FIELD, getUpdateDescription().toBsonDocument());
+    }
+
+    if (stitchDocumentVersion != null) {
+      asDoc.put(Fields.STITCH_DOCUMENT_VERSION_FIELD, stitchDocumentVersion.toBsonDocument());
+    }
+
+    if (stitchDocumentHash != null) {
+      asDoc.put(Fields.STITCH_DOCUMENT_HASH_FIELD, new BsonInt64(stitchDocumentHash));
+    }
+
+    asDoc.put(Fields.WRITE_PENDING_FIELD, new BsonBoolean(hasUncommittedWrites()));
+    return asDoc;
+  }
+
+  /**
+   * Deserializes a {@link BsonDocument} into an instance of change event.
+   * @param document the serialized document
+   * @return the deserialized change event
+   */
+  public static CompactChangeEvent fromBsonDocument(final BsonDocument document) {
+    keyPresent(Fields.OPERATION_TYPE_FIELD, document);
+    keyPresent(Fields.DOCUMENT_KEY_FIELD, document);
+
+    final BsonDocument fullDocument;
+    if (document.containsKey(Fields.FULL_DOCUMENT_FIELD)) {
+      final BsonValue fdVal = document.get(Fields.FULL_DOCUMENT_FIELD);
+      if (fdVal.isDocument()) {
+        fullDocument = fdVal.asDocument();
+      } else {
+        fullDocument = null;
+      }
+    } else {
+      fullDocument = null;
+    }
+
+    final UpdateDescription updateDescription;
+    if (document.containsKey(Fields.UPDATE_DESCRIPTION_FIELD)) {
+      updateDescription = UpdateDescription.fromBsonDocument(
+          document.getDocument(Fields.UPDATE_DESCRIPTION_FIELD)
+      );
+    } else {
+      updateDescription = null;
+    }
+
+    final DocumentVersionInfo.Version stitchDocumentVersion;
+    if (document.containsKey(Fields.STITCH_DOCUMENT_VERSION_FIELD)) {
+      stitchDocumentVersion = DocumentVersionInfo.Version
+          .fromBsonDocument(document.getDocument(Fields.STITCH_DOCUMENT_VERSION_FIELD));
+    } else {
+      stitchDocumentVersion = null;
+    }
+
+    final Long stitchDocumentHash;
+    if (document.containsKey(Fields.STITCH_DOCUMENT_HASH_FIELD)) {
+      stitchDocumentHash = document.getInt64(
+          Fields.STITCH_DOCUMENT_HASH_FIELD
+      ).getValue();
+    } else {
+      stitchDocumentHash = null;
+    }
+
+    return new CompactChangeEvent<>(
+        OperationType.fromRemote(document.getString(Fields.OPERATION_TYPE_FIELD).getValue()),
+        fullDocument,
+        document.getDocument(Fields.DOCUMENT_KEY_FIELD),
+        updateDescription,
+        stitchDocumentVersion,
+        stitchDocumentHash,
+        document.getBoolean(Fields.WRITE_PENDING_FIELD, BsonBoolean.FALSE).getValue());
+  }
```

Comment:
It would be interesting to add some kind of utility function here for people who want to use the compactChangeEvent and combine it with an existing document since we have that logic somewhere (ie applying the update / deleted fields).

Reply:
I’m not sure what the utility of that would be at the moment, unless we had another source of change events coming in...

Reply:
We chatted about this offline, but say you have a Todo item and one of its fields is updated and you want to limit your network usage so you use a consiseChangeEvent. I was just suggesting that the consciseChangeEvent could have a method that takes an existing document and applied the changes in updateFields and deletedFields to the given document.